lib/Basic/FileSystem.cpp

```diff
@@ -14,12 +14,99 @@
 #include "llbuild/Basic/PlatformUtility.h"
 
 #include "llvm/ADT/STLExtras.h"
+#include "llvm/Support/FileSystem.h"
 #include "llvm/Support/Path.h"
 #include "llvm/Support/MemoryBuffer.h"
 
 #include <cassert>
 #include <cstring>
 
+// Cribbed from llvm, where it's been since removed.
+namespace {
+  using namespace std;
+  using namespace llvm;
+  using namespace llvm::sys::fs;
+
+  static std::error_code fillStatus(int StatRet, const struct stat &Status,
+                                    file_status &Result) {
```

Comment:
Yah, I agree. It actually has a small change compared to the original, because that never set `file_type::symlink_file` :)